lib/classes/board.js

```diff
@@ -83,5 +83,17 @@ GridGame.classes.board = function () {
 
   };
 
+  this.place_random_rock = function() {
+    var x = Math.floor(Math.random() * GridGame.width);
+    var y = Math.floor(Math.random() * GridGame.height);
+    var tile = this.tile(x, y);
+    if (tile.city || tile.rock) {
+      this.place_random_rock();
+      return false;
```

Comment:
In that particular scenario, the return may be unnecessary.  If there's no further code after that block, you just have a simple if/else branch.